src/views/primary-auth/CustomButtons.js

```diff
@@ -56,21 +55,19 @@ define([
     },
 
     _createButton: function (options) {
-      var type = options.type.toLowerCase(),
-          attr = 'social-auth-' + type + '-button';
-
+      if(options.isSocialAuth) {
+        options.click =  function(e) {
```

Comment:
Why not just do the click function in settings.js? In general, better to have things look the same - in this case, SocialAuth is using the generic CustomButtons class, rather than having these edge cases built into this generic view.